config/initializers/gitlab_shell_secret_token.rb

```diff
@@ -14,6 +13,8 @@
   File.write(secret_file, token)
 end
 
-if File.exist?(Gitlab.config.gitlab_shell.path) && !File.exist?(gitlab_shell_symlink)
-  FileUtils.symlink(secret_file, gitlab_shell_symlink)
+# TODO May be removed after updating the default secret_file path...?
```

Comment:
I understand your confusion there. This was done because we cannot assume that the users who run gitlab and gitlab-shell has access to the directory where the file resides.